org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/ClientCapabilitiesWrapper.java

```diff
@@ -0,0 +1,87 @@
+package org.eclipse.lsp4xml.settings.capabilities;
+
+import org.eclipse.lsp4j.ClientCapabilities;
+import org.eclipse.lsp4j.DynamicRegistrationCapabilities;
+import org.eclipse.lsp4j.TextDocumentClientCapabilities;
+
+/**
+ * Determines if a client supports a specific capability dynamically
+ */
+public class ClientCapabilitiesWrapper {
+	private boolean dynamicRegistrationSupported;
+
+	public ClientCapabilities capabilities;
+
+	public ClientCapabilitiesWrapper(ClientCapabilities capabilities, boolean dynamicRegistrationSupported) {
```

Comment:
not sure if capabilities can be null, depending on the client, so you might need to add some null-checks